sway-core/src/ir_generation/function.rs

```diff
@@ -128,40 +128,40 @@ impl<'te> FnCompiler<'te> {
         &mut self,
         context: &mut Context,
         md_mgr: &mut MetadataManager,
-        ast_node: ty::TyAstNode,
+        ast_node: &ty::TyAstNode,
     ) -> Result<Option<Value>, CompileError> {
         let span_md_idx = md_mgr.span_to_md(context, &ast_node.span);
-        match ast_node.content {
+        match &ast_node.content {
             ty::TyAstNodeContent::Declaration(td) => match td {
                 ty::TyDeclaration::VariableDeclaration(tvd) => {
-                    self.compile_var_decl(context, md_mgr, *tvd, span_md_idx)
+                    self.compile_var_decl(context, md_mgr, tvd, span_md_idx)
                 }
                 ty::TyDeclaration::ConstantDeclaration(decl_id) => {
-                    let tcd = declaration_engine::de_get_constant(decl_id, &ast_node.span)?;
+                    let tcd = declaration_engine::de_get_constant(decl_id.clone(), &ast_node.span)?;
```

Comment:
I left declaration engine API intact to not cause a massive breaking change, but because of that I had to clone declaration ids. I think it should be fine since we were `clone` ing entire nodes before this PR but if it is preferred to have declaration engine API accepting borrowed types as well I can continue there too.

Reply:
It's a `usize` and a `Span`, not ideal but not the end of the world.